### DISCUSSION TITLE
Continuous integration

### DISCUSSION THREAD

**Discussion by maggie44** [kube-hetzner/terraform-hcloud-kube-hetzner#864] [Q&A]
I am having a think about the best way to do continuous integration with the Terraform deployments and GitHub actions. Would be interested to hear how others have approached it. 

In particular I am looking to have a staging environment and a production. I’m thinking the ideal scenario would be pull requests deploy the code to the staging infrastructure and merges to the production. Can the whole Terraform code, lock files and everything be used for two different instances of the same infrastructure? 

Then there is kubectl apply, maybe in the same repository and to be deployed after terraform commands succeed.

**Comment by mysticaltech** [comment]
@maggie44 Maybe use terraform cloud? See the readme.
